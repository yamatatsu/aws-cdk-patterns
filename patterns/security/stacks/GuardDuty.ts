import {
  App,
  Stack,
  StackProps,
  aws_guardduty as guardduty,
  aws_events as events,
  aws_events_targets as events_targets,
  aws_sns as sns,
} from "aws-cdk-lib"

export class GuardDuty extends Stack {
  public readonly topic: sns.ITopic

  constructor(scope: App, id: string, props?: StackProps) {
    super(scope, id, props)

    new guardduty.CfnDetector(this, "GuardDuty", {
      enable: true,
    })

    const topic = new sns.Topic(this, "GuardDutyTopic")

    const rule = new events.Rule(this, "GuardDutyNotificationRule", {
      ruleName: "GuardDutyNotificationRule",
      description: "Alert to SNS topic when find threats by Guardduty",
      eventPattern: {
        source: ["aws.guardduty"],
        detailType: ["GuardDuty Finding"],
      },
    })
    rule.addTarget(new events_targets.SnsTopic(topic))

    this.topic = topic
  }
}
