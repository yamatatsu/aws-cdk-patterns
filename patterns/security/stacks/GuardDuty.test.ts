import { SynthUtils } from "@aws-cdk/assert"
import * as cdk from "@aws-cdk/core"
import { GuardDuty } from "./GuardDuty"

test("snapshot test", () => {
  const app = new cdk.App()

  const target = new GuardDuty(app, "Target")

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot()
})
