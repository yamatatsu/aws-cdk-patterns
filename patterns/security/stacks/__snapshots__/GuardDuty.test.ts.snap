// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "GuardDuty": Object {
      "Properties": Object {
        "Enable": true,
      },
      "Type": "AWS::GuardDuty::Detector",
    },
    "GuardDutyNotificationRule6F3FBB81": Object {
      "Properties": Object {
        "Description": "Alert to SNS topic when find threats by Guardduty",
        "EventPattern": Object {
          "detail-type": Array [
            "GuardDuty Finding",
          ],
          "source": Array [
            "aws.guardduty",
          ],
        },
        "Name": "GuardDutyNotificationRule",
        "State": "ENABLED",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Ref": "GuardDutyTopic003A9A5B",
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "GuardDutyTopic003A9A5B": Object {
      "Type": "AWS::SNS::Topic",
    },
    "GuardDutyTopicPolicy0CA49850": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sns:Publish",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "events.amazonaws.com",
              },
              "Resource": Object {
                "Ref": "GuardDutyTopic003A9A5B",
              },
              "Sid": "0",
            },
          ],
          "Version": "2012-10-17",
        },
        "Topics": Array [
          Object {
            "Ref": "GuardDutyTopic003A9A5B",
          },
        ],
      },
      "Type": "AWS::SNS::TopicPolicy",
    },
  },
}
`;
